sdk/src/main/java/co/omise/android/ThreeDS2ServiceWrapper.kt

```diff
@@ -26,21 +32,56 @@ internal class ThreeDS2ServiceWrapper(
     lateinit var transaction: Transaction
         private set
 
-    suspend fun initialize() =
+    // Computes SHA-512 hash of a string
+    fun hash512(data: String): ByteArray {
+        val hasher = MessageDigest.getInstance("SHA-512")
+        return hasher.digest(data.toByteArray(Charsets.UTF_8))
+    }
+
+    // Decrypts the encrypted data using the decryption key and AES/CTR/NoPadding cipher
+    fun aesDecrypt(
+        ciphertext: ByteArray,
```

Comment:
Should we consider creating a dedicated class for `hash512` and `aesDecrypt`?

Reply:
If we were also encrypting stuff i would say we should consider that, but the usage is pretty limited and centered around 3DS.

Reply:
Although the usage is limited at the moment, we now have 2 reasons to update the class which violates SRP. One related to encryption and another related to 3DS.

Separation ensures that the primary class remains focused on its core responsibility of handling 3DS-specific logic, resulting in improved clarity and better comply with SRP.